runtime/src/identity.rs

```diff
@@ -844,6 +859,18 @@ impl<T: Trait> Module<T> {
             <CurrentDid>::kill();
         }
     }
+    /// It adds `skey` to pre authorized keys for `id` identity.
+    fn add_pre_join_identity(skey: &SigningKey, id: IdentityId) {
```

Comment:
The front-end will likely have to fetch the list of pre-auths for a given account so the user can choose which one to join. Can this list be easily accessed from outside? 
Would be nice to store the date the entry was added as well for the front-end to display.

Reply:
I think it will be best if the middleware layer stores that info.

Reply:
So the options look like:

1. use middleware (e.g. graphQL) to track events to reconstruct what DIDs have pre-authorised what keys. @pabloruiz55 's point is that this would make 3rd party UIs harder (as they would need to rely on our middleware or build their own).

2. add something like:
```
        /// Pre-authorize join to Identity.
        pub PreAuthorizedJoinDids get(pre_authorized_join_dids): map Key => Vec<IdentityId>;
```
to store the full vector of DIDs which have pre-authorised a Key. To avoid having to loop through this vector to remove an IdentityId when it is authorised, we could add to the `PreAuthorizedKeyInfo` struct a `IndexInPreAuthorizedJoinDids` value which is the index of the DID in `PreAuthorizedJoinDids` for a particular account (so that it can be easily found in the vector without looping).

Reply:
Option 2. does create "duplicated" data, but in this case, given the data is likely to be small and get cleaned up along the way, this seems reasonable. @maxsam4 any concerns with this approach?

Reply:
I might be wrong but I think that on substrate, the whole vector has to be read and deserialized to access even a single element. That's why storing an index won't really help. That's also why I am a little "anti-vectors".

ps. We can open-source our middleware and anyone will be able to run it just like they can run polymesh nodes.

Reply:
Substrate offers "linked map" that we can use instead of vector. They are used just like "map" but they behave like linked lists and UI can query all elements in a linked map.